packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -516,16 +517,25 @@ class IrisGridTableModel extends IrisGridModel {
   get columns() {
     return this.getColumnsWithLayoutHints(
       this.table.columns,
-      this.table.layoutHints
+      this.layoutHints,
+      this.frozenColumns
     );
   }
 
+  get frozenColumns() {
+    if (this.userFrozenColumns && this.userFrozenColumns != null) {
+      return this.userFrozenColumns;
+    }
+
+    return this.layoutHints?.frozenColumns ?? [];
```

Comment:
We should either memoize this (similar to `get columns()`, e.g. `getFrozenColumnsWithLayoutHints`), or have a static empty array created that is used here. Otherwise any UI looking at `model.frozenColumns` may think it's changed if just doing a shallow equals